tests/unit/Vision/VisionHelpersTraitTest.php

```diff
@@ -0,0 +1,144 @@
+        $features = [$feature];
+        $imageContext = new ImageContext();
+        $imageContext->setLanguageHints(['en']);
+
+        $cb = function ($requests, $optionalArgs) use ($image, $features, $imageContext) {
+
+            // Test that imageContext key is correctly stripped
+            $this->assertArrayNotHasKey('imageContext', $optionalArgs);
+
+            $this->assertSame(1, count($requests));
+            $request = $requests[0];
+            $this->assertEquals($image, $request->getImage());
+            // Use iterator_to_array to convert protobuf Repeated Field object to array for comparison
+            $this->assertSame($features, iterator_to_array($request->getFeatures()));
+            $this->assertSame($imageContext, $request->getImageContext());
+
+            $response = new BatchAnnotateImagesResponse();
+            $response->setResponses([new AnnotateImageResponse()]);
+            return $response;
+        };
+
+        $response = $this->implementation->call('annotateImageHelper', [
+            $cb,
+            AnnotateImageRequest::class,
+            $image,
+            $features,
+            ['imageContext' => $imageContext]
+        ]);
+
+        $this->assertEquals(AnnotateImageResponse::class, get_class($response));
+    }
+
+    /**
+     * @dataProvider createImageHelperDataProvider
+     */
+    public function testCreateImageHelper($imageInput, $expectedContent, $expectedUri)
+    {
+        $image = $this->implementation->call('createImageHelper', [
+            Image::class,
+            ImageSource::class,
+            $imageInput
+        ]);
+        $this->assertSame($expectedContent, $image->getContent());
+        $imageSource = $image->getSource();
+        $this->assertSame($expectedUri, is_null($imageSource) ? null : $imageSource->getImageUri());
+    }
+
+    public function createImageHelperDataProvider()
+    {
+        $content = 'imageresourcecontent';
+        $stream = fopen('php://memory','r+');
+        fwrite($stream, $content);
+        rewind($stream);
+        return [
+            ["http://my.site/myimage.jpg", "", "http://my.site/myimage.jpg"],
+            ["https://my.site/myimage.jpg", "", "https://my.site/myimage.jpg"],
+            ["gs://my_bucket/myimage.jpg", "", "gs://my_bucket/myimage.jpg"],
+            ["abcdefxyz", "abcdefxyz", null],
+            [$stream, $content, null],
+        ];
+    }
+
+    /**
+     * @dataProvider invalidCreateImageHelperDataProvider
+     * @expectedException InvalidArgumentException
+     */
+    public function testInvalidCreateImageHelper($imageInput)
+    {
+        $this->implementation->call('createImageHelper', [
+            Image::class,
+            ImageSource::class,
+            $imageInput
+        ]);
+    }
+
+    public function invalidCreateImageHelperDataProvider()
+    {
+        return [
+            [null],
+            [new Image()],
+            [5],
+        ];
+    }
+}
+
+class VisionHelpersTraitStub
```

Comment:
You could use `Google\Cloud\Core\TestHelpers::impl()` instead of a stub class in this case.

Reply:
Done

Reply:
this class can be dropped now.

Reply:
I have dropped the class. But I found that i needed to add a `use VisionHelpersTrait` to the test class, otherwise I would get an error. Is this normal?